plugins/lookup/bitwarden.py

```diff
@@ -142,45 +146,46 @@ def _run(self, args, stdin=None, expected_rc=0):
             raise BitwardenException(err)
         return to_text(out, errors='surrogate_or_strict'), to_text(err, errors='surrogate_or_strict')
 
-    def _get_matches(self, search_value, search_field, collection_id=None):
+    def _get_matches(self, search_value, search_field, collection_id=None, organization_id=None):
         """Return matching records whose search_field is equal to key.
         """
 
         # Prepare set of params for Bitwarden CLI
-        if search_value:
-            if search_field == 'id':
-                params = ['get', 'item', search_value]
-            else:
-                params = ['list', 'items', '--search', search_value]
-            if collection_id:
-                params.extend(['--collectionid', collection_id])
+        if search_field == 'id':
+            params = ['get', 'item', search_value]
         else:
-            if not collection_id:
-                raise AnsibleError("search_value is required if collection_id is not set.")
-
-            params = ['list', 'items', '--collectionid', collection_id]
+            params = ['list', 'items', '--search', search_value if search_value else '']
 
```

Comment:
Why no longer add `--collectionid`? Shouldn't that make the call more efficient since it potentially returns less results?

Reply:
`--collectionid` is "mutually exclusive" with `--organizationid`, since `bw` ignores `--organizationid` when `--collectionid` is used: https://github.com/bitwarden/clients/issues/2736
At the same time both options just filter a json dict by their top level attributes, e.g. a very simple task.

Therefore, I opted for just doing that in the lookup, instead of trying to do workarounds in the lookup and either
- checking 4 cases of combinations of these options and sometimes filtering twice or 
- propagating short commings of the CLI which are a one-liner to implement.

Do you agree with that assessment?

Reply:
I would still also supply these options, to minimize the data to be processed as early as possible.

Reply:
There is a`bw list org-collections`, which will probably filter when both orgid and collid are set, so we could differentiate between 4 cases.

But, as a last question:
I checked that `bw` also does all of its filtering locally in javascript (https://github.com/bitwarden/cli/blob/master/src/commands/list.command.ts#L56), so there is no optimization skipped by not asking the server to filter or something.
Is your opinion on minimizing the data as early as possible in this case based on "lets filter in the first loop inside the bw-javascript, to have the second ansible-python loop smaller"? Or was it maybe the assumption that the `bw` tool might do something more efficient than our python-loop?

I will follow your lead and add the necessary code, just wanted to make sure we have all the information first.

Reply:
> But, as a last question: I checked that `bw` also does all of its filtering locally in javascript (https://github.com/bitwarden/cli/blob/master/src/commands/list.command.ts#L56), so there is no optimization skipped by not asking the server to filter or something. Is your opinion on minimizing the data as early as possible in this case based on "lets filter in the first loop inside the bw-javascript, to have the second ansible-python loop smaller"? Or was it maybe the assumption that the `bw` tool might do something more efficient than our python-loop?

The `bw` tool does that right now. At some point someone might optimize this, and suddenly filtering happens on the server side. If you assume this behavior will never change, and don't pass on filtering parameters to the `bw` tool, you'll miss that free optimization.

This could also become important at some point if the server records an audit log of who accessed what; if the server then starts doing the filtering (which it only can if the `bw` tool knows about the filtering) the audit logs are more precise.

Obviously if that optimization never gets added nothing is gained. But you never know :)

Reply:
That's a good argument and I agree.

Also, I seem to have misunderstood collections. It seems collections always belong to an organization (or your personal vault), so it makes sense to ignore `organizationid` in that case since the collection already implicitly specifies the organization.

Therefore, this change is way smaller and just adds a parameter and some refactoring. If there is nothing else left, you can continue rebasing and merging to your liking.